src/libraries/System.Data.Common/src/System/Data/SQLTypes/SQLGuid.cs

```diff
@@ -292,7 +282,7 @@ public int CompareTo(SqlGuid value)
             (this == other).Value;
 
         // For hashing purpose
-        public override int GetHashCode() => IsNull ? 0 : Value.GetHashCode();
+        public override int GetHashCode() => IsNull ? 0 : _value.GetValueOrDefault().GetHashCode();
```

Comment:
`default(Guid?).GetHashCode()` is already zero so instead of checking `IsNull` explicitly I think this could just be `=> _value.GetHashCode();`